### PR TITLE
security

### PR DESCRIPTION
**Pull request by VictorGSoutoXP**
This PR adds improvements to the LLM-generated code execution code in Python. Added type checking and type conversion capabilities, as well as restricting access to modules and functions deemed unsafe through the use of Python's ast library. In addition, unit tests were performed to ensure the correct functioning of the code. These improvements are expected to increase the security and robustness of code generated by LLM in Python.

### CODE REVIEW

File: pandasai/llm/security.py
Comment (gventuri) [inline pandasai/llm/security.py:1]: Hey @VictorGSoutoXP could you move this to the helpers folder?
Also, could you also implement this in the LLMs?
Could you also add `matplotlib` to the allowed libraries?
I'm also a little bit concerned of whitelisting only those 3 methods, what do you think?

Finally, as this is a very sensitive change, we should probably enforce it with a few tests, what do you think?

Thanks a lot for the contribution!!

**Review by gventuri — CHANGES_REQUESTED** [review]

**Comment by gventuri** [comment]
Hey @VictorGSoutoXP, any updates?

**Comment by avelino** [comment]
relate PR to issue #43 
People who participated in the discussion in the issue can understand what is being done

**Comment by gventuri** [comment]
Closing, as the error has been fixed! Thanks @VictorGSoutoXP once again for the support!

**Pull request closed without merging**